core/api/src/mill/api/Watchable.scala

```diff
@@ -8,6 +8,24 @@ package mill.api
  */
 private[mill] sealed trait Watchable
 private[mill] object Watchable {
-  case class Path(p: java.nio.file.Path, quick: Boolean, signature: Int) extends Watchable
-  case class Value(f: () => Long, signature: Long, pretty: String) extends Watchable
+
+  /** A [[Watchable]] that is being watched via polling. */
+  private[mill] sealed trait Pollable extends Watchable
+
+  /** A [[Watchable]] that is being watched via a notification system (like inotify). */
+  private[mill] sealed trait Notifiable extends Watchable
```

Comment:
Do we actually need these two additional marker traits? It seems to me that working directly with `Watchable.Path` and `Watchable.Value` will be enough. There aren't any compatibility/extensibility concerns since all this stuff is `private[mill]` anyway